schemachangerccl: TestBackupbase_add_column_default_seq is flaky

**Describe the problem**

Observed in [this CI run](https://teamcity.cockroachdb.com/viewLog.html?buildId=6583066&buildTypeId=Cockroach_UnitTests).

```console
=== RUN   TestBackupbase_add_column_default_seq/ALTER_TABLE_db.public.tbl_ADD_COLUMN_l_INT_NOT_NULL_DEFAULT_nextval('db.public.sq1')/backup/restore_stage_3_of_10
    cumulative.go:506: stage 1/10 in PostCommitPhase (rollback=false) 3 "db" false
    cumulative.go:506: stage 2/10 in PostCommitPhase (rollback=false) 3 "db" false
    cumulative.go:506: stage 3/10 in PostCommitPhase (rollback=false) 3 "db" false
    cumulative.go:506: stage 4/10 in PostCommitPhase (rollback=false) 3 "db" false
    cumulative.go:506: stage 1/2 in PostCommitNonRevertiblePhase (rollback=true) 3 "db" false
    cumulative.go:506: stage 2/2 in PostCommitNonRevertiblePhase (rollback=true) 3 "db" true
    cumulative.go:793: condition failed to evaluate within 45s: waiting for statements to execute
    cumulative.go:544: 
        	Error Trace:	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/schemachangerccl/cumulative.go:544
        	            				/go/src/github.com/cockroachdb/cockroach/pkg/ccl/schemachangerccl/cumulative.go:654
        	Error:      	Expect "<nil>" to match "boom 3"
        	Test:       	TestBackupbase_add_column_default_seq/ALTER_TABLE_db.public.tbl_ADD_COLUMN_l_INT_NOT_NULL_DEFAULT_nextval('db.public.sq1')/backup/restore_stage_3_of_10
        --- FAIL: TestBackupbase_add_column_default_seq/ALTER_TABLE_db.public.tbl_ADD_COLUMN_l_INT_NOT_NULL_DEFAULT_nextval('db.public.sq1')/backup/restore_stage_3_of_10 (68.78s)
=== RUN   TestBackupbase_add_column_default_seq/ALTER_TABLE_db.public.tbl_ADD_COLUMN_l_INT_NOT_NULL_DEFAULT_nextval('db.public.sq1')
    cumulative.go:645: test case has 7 revertible post-commit stages and 3 non-revertible post-commit stages
    --- FAIL: TestBackupbase_add_column_default_seq/ALTER_TABLE_db.public.tbl_ADD_COLUMN_l_INT_NOT_NULL_DEFAULT_nextval('db.public.sq1') (137.95s)
=== RUN   TestBackupbase_add_column_default_seq
    test_log_scope.go:161: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestBackupbase_add_column_default_seq3354608072
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestBackupbase_add_column_default_seq
    panic.go:522: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestBackupbase_add_column_default_seq3354608072
--- FAIL: TestBackupbase_add_column_default_seq (138.11s)
```

**To Reproduce**

Attempted to reproduce locally:

```bash
$ ./dev test ./pkg/ccl/schemachangerccl --filter TestBackupbase_add_column_default_seq --stress
```

Jira issue: CRDB-19875

--- a/pkg/sql/schemachanger/scrun/scrun.go
+++ b/pkg/sql/schemachanger/scrun/scrun.go
@@ -158,13 +158,15 @@ func executeStage(
 	stageIdx int,
 	stage scplan.Stage,
 ) (err error) {
+
+	log.Infof(ctx, "executing %s (rollback=%v)", stage, p.InRollback)
+
 	if knobs != nil && knobs.BeforeStage != nil {
 		if err := knobs.BeforeStage(p, stageIdx); err != nil {
 			return err
 		}
 	}
 
-	log.Infof(ctx, "executing %s (rollback=%v)", stage, p.InRollback)
 	start := timeutil.Now()
 	defer func() {
 		if log.ExpensiveLogEnabled(ctx, 2) {

--- a/pkg/sql/schemachanger/sctest/cumulative.go
+++ b/pkg/sql/schemachanger/sctest/cumulative.go
@@ -534,6 +534,10 @@ func Backup(t *testing.T, path string, newCluster NewClusterFunc) {
 			if s.p.InRollback {
 				rollbackStage++
 			}
+			if done {
+				t.Logf("reached final stage, waiting for completion")
+				stageChan = nil // allow the restored jobs to proceed
+			}
 			if shouldFail {
 				s.resume <- errors.Newf("boom %d", i)
 			} else {
@@ -545,7 +549,7 @@ func Backup(t *testing.T, path string, newCluster NewClusterFunc) {
 		} else {
 			require.NoError(t, err)
 		}
-		stageChan = nil // allow the restored jobs to proceed
+
 		t.Logf("finished")
 
 		for i, b := range backups {

